### ISSUE TITLE
HTTP service hangs when a http call is made inside a resource

### ISSUE THREAD

**Issue by chalitha1989** [ballerina-platform/ballerina-standard-library#3650]
**Description:**

$subject. Please see the code and below steps to reproduce.

**Steps to reproduce:**

1. Create a service using below code and then build and run.

```ballerina
import ballerina/http;


service / on new http:Listener(9090) {


    resource function get greeting() returns string|error {

         final http:Client clientEndpoint = check new ("https://run.mocky.io/v3/164eda0f-9980-4aa1-9e09-94d8ac6ca6e0");
         json resp = check clientEndpoint->get("/");
         return "Resp: " + resp.toBalString();
    }
}
```

2. Execute below curl to invoke the service for the first time:

`curl localhost:9090/greeting`

3. Execute the curl for the second time. The client and the server hangs. 

<img width="404" alt="Screenshot 2022-11-14 at 23 37 16" src="https://user-images.githubusercontent.com/13028527/201734141-336eb911-8837-46e9-8727-781af6a27c38.png">

**Affected Versions:**

**OS, DB, other environment details and versions:**

Ballerina - 2201.2.3
Http - 2.4.3
Mac OS - 12.0.1 (21A559)

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by chamil321** [comment]
This occurs when client HTTP protocol is HTTP2 and backend service is HTTPS. As a workaround either you can downgrade the client to use HTTP1.1 or change the backend service to HTTP
We'll fix this and do a patch on 2201.2.x path

**Comment by chamil321** [comment]
Fix has been released to DEV CENTRAL for choreo testing today

**Comment by chamil321** [comment]
Released HTTP v2.4.4

**Comment by chamil321** [comment]
Released HTTP v2.4.4 https://github.com/ballerina-platform/module-ballerina-http/releases/tag/v2.4.4
